### PR TITLE
Fix cookie branch

### PR DESCRIPTION
**Pull request by Karyum**
So the cookie is being set right, apparently you do [need the `withCredentials` to be true](https://stackoverflow.com/questions/43002444/make-axios-send-cookies-in-its-requests-automatically).

and you also need the cors middleware to be configured to accept any origins and the credentials header [click here](https://expressjs.com/en/resources/middleware/cors.html#configuration-options)

@LogicX1 i'm assigning you so you can how we solved it.

### CODE REVIEW

**Comment by Karyum** [comment]
mistakes were made, my bad, this branch is supposed to be a pull request to @hadeel11 's branch `cookie`

File: server/src/app.js [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
 require('dotenv').config();
 
 const app = express();
 
 app.set('port', process.env.PORT || 4000);
-app.use(cors());
+app.use(cookieParser());
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
+app.use(cors({origin:true,credentials: true}));
```
Comment (LogicX1) [inline server/src/app.js:15]: I see glad it worked, however with this cors configuration , anyone can use your server to get/post data ?

Comment (Karyum) [reply server/src/app.js:15]: @LogicX1 close your eyes when you delve into this PR, it's chaos

Comment (Karyum) [reply server/src/app.js:15]: > I see glad it worked, however with this cors configuration , anyone can use your server to get/post data ?

technically yes, think about it as an open API, you just have to secure certain routes that you want secured, that's why we have the `authCheck` middleware

Comment (LogicX1) [reply server/src/app.js:15]: I see well if it works it works!
We were saved of this hassle by using auth0, though I still wasted a lot of time reading their documentation and tutorials

**Pull request closed without merging**